cli/container.go

```diff
@@ -74,17 +150,14 @@ func createServer(verbose bool, image string, port string, args []string, env []
 
 	containerName := fmt.Sprintf("k3d-%s-server", name)
 
-	containerPort := nat.Port(fmt.Sprintf("%s/tcp", port))
+	apiPortSpec := fmt.Sprintf("0.0.0.0:%s:6443/tcp", port)
```

Comment:
As requested in the related issue, people might want to map also the APIServer port to a different IP than 0.0.0.0.
But I wouldn't change it in the scope of this PR, since we're introducing node roles etc. anyways in a follow-up PR :+1:

Reply:
That's about the --api-port stuff we talked about right?  I did not want to increase the scope of this pull request.  Sure, let's track it in the follow up PR.  

This pull request only implements the current --port feature.